egs/jnas/asr1/run.sh

```diff
@@ -0,0 +1,253 @@
+ctc_weight=0.3
+recog_model=model.acc.best # set a model to be used for decoding: 'model.acc.best' or 'model.loss.best'
+
+# scheduled sampling option
+samp_prob=0.0
+
+# data
+jnas_train_root=/database/JNAS                # root directory including original data
+jnas_eval_root=/database/JNAS/DOCS/Test_set   # root directory including test data
+
+# exp tag
+tag="" # tag for managing experiments.
+
+. utils/parse_options.sh || exit 1;
+
+. ./path.sh
+. ./cmd.sh
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+train_set=train_nodev
+train_dev=train_dev
+recog_set="JNAS_testset_100 JNAS_testset_500"
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    ### Task dependent. You have to make data the following preparation part by yourself.
+    ### But you can utilize Kaldi recipes in most cases
+    echo "stage 0: Data Preparation"
+    # Initial normalization of the data
+    local/jnas_train_prep.sh ${jnas_train_root} ./conf/train_speakers.txt
+    for rtask in ${recog_set};do
+        local/jnas_eval_prep.sh ${jnas_eval_root} ${rtask}
+    done
+
+fi
+
+feat_tr_dir=${dumpdir}/${train_set}/delta${do_delta}; mkdir -p ${feat_tr_dir}
+feat_dt_dir=${dumpdir}/${train_dev}/delta${do_delta}; mkdir -p ${feat_dt_dir}
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    ### Task dependent. You have to design training and dev sets by yourself.
+    ### But you can utilize Kaldi recipes in most cases
+    echo "stage 1: Feature Generation"
+    fbankdir=fbank
+    # Generate the fbank features; by default 80-dimensional fbanks with pitch on each frame
+    steps/make_fbank_pitch.sh --cmd "$train_cmd" --nj 10 --write_utt2num_frames true \
+        data/train exp/make_fbank/train ${fbankdir}
+
+    # remove utt having more than 2000 frames or less than 10 frames or
+    # remove utt having more than 200 characters or 0 characters
+    remove_longshortdata.sh data/train data/train_trim
+
+    # following split consider prompt duplication (but does not consider speaker overlap instead)
+    local/split_tr_dt.sh --perdt 5 data/train_trim \
+	    data/${train_set} data/${train_dev}
+    
+    # compute global CMVN
+    compute-cmvn-stats scp:data/${train_set}/feats.scp data/${train_set}/cmvn.ark
+
+    dump.sh --cmd "$train_cmd" --nj 16 --do_delta ${do_delta} \
+        data/${train_set}/feats.scp data/${train_set}/cmvn.ark exp/dump_feats/train ${feat_tr_dir}
+
+    dump.sh --cmd "$train_cmd" --nj 4 --do_delta ${do_delta} \
+        data/${train_dev}/feats.scp data/${train_set}/cmvn.ark exp/dump_feats/dev ${feat_dt_dir}
+
+    for rtask in ${recog_set}; do
+	# Generate the fbank features.
+	steps/make_fbank_pitch.sh --cmd "$train_cmd" --nj 10 --write_utt2num_frames true \
+	    data/${rtask} exp/make_fbank/${rtask} ${fbankdir}
+
+	# remove longer or shorter datas.
+	remove_longshortdata.sh data/${rtask} data/${rtask}_trim
```

Comment:
For recog sets, we should not perform `remove_longshortdata.sh`.
(Maybe `${rtask}_trim` is not used?)